target-chains/cosmwasm/tools/src/deploy-pyth-bridge.ts

```diff
@@ -1,115 +1,181 @@
+  .parse();
 
-// checks
-if (inputCodeId === undefined && artifact === undefined)
-  raiseCLError("Please provide either artifact or code id");
-
-function sleep(ms: number) {
-  return new Promise((resolve) => setTimeout(resolve, ms));
+function getDeployer(network: NETWORKS, mnemonic: string): Deployer {
+  const networkConfig = NetworkConfig[network];
+  return DeployerFactory.create(networkConfig, mnemonic);
 }
 
-function raiseCLError(message: string) {
-  console.error(message);
-  process.exit(1);
+async function deployArtifactHandler(argv: any) {
+  const { network, mnemonic, artifact } = argv;
+  console.log("Deploying artifact");
+  const codeId = await getDeployer(
+    network as NETWORKS,
+    mnemonic as string
+  ).deployArtifact(artifact);
+  console.log("Deployed Code ID: ", codeId);
 }
 
-async function run() {
-  let codeId: number;
-  if (inputCodeId === undefined) {
-    console.log("Deploying artifact");
-    codeId = await deployer.deployArtifact(artifact!);
-    console.log("Deployed Code ID: ", codeId);
+async function instantiateHandler(argv: any) {
+  const { network, mnemonic, codeId } = argv;
+  const pythConfig = PythConfig[network as NETWORKS];
+  console.log("Instantiating a contract");
+  const contractAddress = await getDeployer(
+    network as NETWORKS,
+    mnemonic as string
+  ).instantiate(codeId, pythConfig, "pyth");
+  console.log(`Deployed Pyth contract at ${contractAddress}`);
+}
 
-    // sleep only when a new artifact is deployed
-    if (instantiate || migrate) {
-      console.log("Sleeping for 10 seconds for store transaction to finalize.");
-      await sleep(10000);
-    }
-  } else codeId = inputCodeId;
+async function migrateHandler(argv: any) {
+  const { network, mnemonic, codeId, contract } = argv;
 
-  if (instantiate) {
-    console.log("Instantiating a contract");
-    const contractAddress = await deployer.instantiate(
-      codeId,
-      pythConfig,
-      "pyth"
-    );
-    console.log(`Deployed Pyth contract at ${contractAddress}`);
-  }
-  if (migrate) {
-    if (contract === "")
-      raiseCLError(
-        "Contract address is not provided. Provide it using --contract"
-      );
+  console.log(`Migrating contract ${contract} to ${codeId}`);
+  await getDeployer(network as NETWORKS, mnemonic as string).migrate(
+    contract,
+    codeId
+  );
+  console.log(`Contract ${contract} code_id successfully updated to ${codeId}`);
+}
 
-    console.log(`Migrating contract ${contract} to ${codeId}`);
-    await deployer.migrate(contract, codeId);
-    console.log(
-      `Contract ${contract} code_id successfully updated to ${codeId}`
-    );
-  }
+async function updateAdminHandler(argv: any) {
+  const { network, mnemonic, newAdmin, contract } = argv;
+  console.log("Updating contract's admin");
+  await getDeployer(network as NETWORKS, mnemonic as string).updateAdmin(
+    newAdmin,
+    contract
+  );
+  console.log("Contract's admin successfully updates");
```

Comment:
Is this for removing the admin? can't we have it setAdminToZero? or something like this?

Reply:
on cosmos, you set the contract's admin address to itself, which allows the contract to send itself the migrate message.